sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/FileIndexer.java

```diff
@@ -120,20 +123,32 @@ public void index() {
     }
   }
 
-  private void waitForTasksToComplete() {
+  private void waitForTasksToComplete(ProgressReport report) {
     executorService.shutdown();
     for (Future<Void> task : tasks) {
       try {
         task.get();
       } catch (ExecutionException e) {
         // Unwrap ExecutionException
+        stopAsap(report);
         throw e.getCause() instanceof RuntimeException ? (RuntimeException) e.getCause() : new IllegalStateException(e.getCause());
       } catch (InterruptedException e) {
+        stopAsap(report);
         throw new IllegalStateException(e);
       }
     }
   }
 
+  private void stopAsap(ProgressReport report) {
+    report.stop(null);
+    executorService.shutdownNow();
+    try {
+      executorService.awaitTermination(5, TimeUnit.SECONDS);
+    } catch (InterruptedException e1) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=improve_index_error_logs&issues=AWCiHi8cqDrCo5ysgHe0&open=AWCiHi8cqDrCo5ysgHe0)

[ISSUE_KEY]: AWCiHi8cqDrCo5ysgHe0